internal/storage/rocksdb/store.go

```diff
@@ -515,7 +592,17 @@ func (rdb *rocksDB) getSingleKey(ro *gorocksdb.ReadOptions, key []byte) ([]*serv
 	defer value.Free()
 	val := toByteArray(value)
 	if val != nil && len(val) > 0 {
-		return []*serverpb.KVPair{&serverpb.KVPair{Key: key, Value: val}}, nil
+		//check ttl.
+		expireTs, _ := getExpireTs(val)
+		if expireTs > 0 && expireTs < time.Now().Unix() {
```

Comment:
In practice, the `time.Now()` is pretty expensive since it involves the kernel. Check if the time-keeping can be done more cheaply, perhaps in user space ?

Reply:
Moved to fastTime.